### ISSUE TITLE
MacOS 12.3 can not use this plugin, how to resolve?

### ISSUE THREAD

**Issue by jankerli** [lukewaite/alfred-gitlab#27]
MacOS 12.3 remove Python2.x, when i install Python 3, but it doesn't work, the error message: 
`-e:1: syntax error, unexpected tSTRING_BEG, expecting do or '{' or '(' python gitlab.py --seturl "`
how to fix it? Thanks~

**Comment by GabLeRoux** [comment]
`/usr/bin/python` doesn't exist anymore on macOS 12.3. This leads to `/bin/bash: python2: command not found` when running the plugin with debug console open.

A possible workaround (definitely not the right solution, but works and good enough if you use it daily as I do) is to locate each occurrence of `/usr/bin/python` in the code and `python` in the and replace them with the path to an installation of `python2.7` that you will manage by yourself.

It would definitely be preferable to update this to python3, but if just want it to work, here are full steps. This assumes you have [`conda`](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html).

```bash
conda create -n alfred-gitlab python=2.7
conda activate alfred-gitlab
which python # this will give you the python path
# /Users/gableroux/miniconda/envs/alfred-gitlab/bin/python
```

Update these:

<img width="922" alt="image" src="https://user-images.githubusercontent.com/1264761/158747467-ef44569c-2d0b-49ce-9f89-b8541dcad6ac.png">

Example:

<img width="831" alt="image" src="https://user-images.githubusercontent.com/1264761/158747720-0f912525-aeab-4e38-80f5-1555c26dd818.png">

Open `Alfred.alfredpreferences/workflows/user.workflow.D7406121-E5D3-438B-81D5-70FF7A9C58AC/` (path to the workflow files) and do a search and replace from `/usr/bin/python` to the new python path.

Here you go, shiny workflow again (manually altered, but good enough until this gets a proper fix).

<img width="613" alt="image" src="https://user-images.githubusercontent.com/1264761/158747871-a99f2bca-6634-4b8a-9514-45b17afff856.png">

**Comment by jankerli** [comment]
ths~

**Comment by SaschaKrug** [comment]
What helped me as a temporary solution:

1. Follow the guide to install python2: [www.alfredapp.com](https://www.alfredapp.com/help/kb/python-2-monterey/)
2. Do `python --version`. Should be python 2.7
3. Get the new path via `which python`. In my case it was `/usr/local/bin/python`
4. Replace `/usr/bin/python` with the new path in the Alfred Workflow and inside the workflow folder. I needed to replace it inside of `workflow/background.py` and `gitlab.py`

Use the Alfred Workflow Debugger in the Preferences for assistance.

And yes, a new workflow would definitely be needed, but I don't have currently time to port it to python3. It looks like there  are not even Alfred helpers for python3? That's weird ...

**Comment by pat-s** [comment]
See https://github.com/deanishe/alfred-workflow/issues/97 and the patched fork at https://github.com/NorthIsUp/alfred-workflow-py3.

**Comment by mjhuber** [comment]
if your main use case is searching for repos, i wrote a workflow in go that can be used until this excellent workflow is updated to support py3: https://github.com/mjhuber/alfred-gitlab

**Comment by TribuneX** [comment]
I upgraded this workflow to be compatible with python3 (with some help of @meinjens):
https://github.com/TribuneX/alfred-gitlab

I upgraded to a python3 compatible alfred-workflow version and replaced the old `web.py` with [mureq](https://github.com/slingamn/mureq). I'll see if I will open a PR here to get these changes included in this workflow.

**Comment by pat-s** [comment]
@TribuneX Awesome!

I've successfully used your fork to port [alfred-gitea](https://github.com/pat-s/alfred-gitea/releases/tag/2.0.0) to Python3.
I was trying since a few days and struggling (as I am not a Python professional) but your forked got me going quickly.

🚀

**Comment by lukewaite** [comment]
Fixed thanks to @TribuneX in [3.0.0](https://github.com/lukewaite/alfred-gitlab/releases/tag/v3.0.0) which you may need to manually install if you're already on 12.3.

**Comment by rtvenge** [comment]
Thank you @TribuneX for updating and to @lukewaite for merging it in so quickly! This whole last week for me was `gl foo… 🤦🏻` 😆
